servant-auth-server/src/Servant/Auth/Server/Internal/ConfigTypes.hs

```diff
@@ -79,7 +75,31 @@ defaultCookieSettings = CookieSettings
     , cookiePath        = Just "/"
     , cookieSameSite    = SameSiteLax
     , sessionCookieName = "JWT-Cookie"
-    , xsrfCookieName    = "XSRF-TOKEN"
+    , xsrfSetting       = XsrfCookie def
+    }
+
+data XsrfSetting = XsrfCookie XsrfCookieSettings
+                 | XsrfCheckOrigin (Maybe BS.ByteString -> Bool)
+                 deriving (Generic)
```

Comment:
Maybe adding two constructors, `XsrfOr XsrfSetting XsrfSetting` and `XsrfAnd XsrfSetting XsrfSetting`, so that we can combine checks? 

Also, `XsrfCheckReferer`, analogous to `XsrfCheckOrigin`, would be nice to add since it's simple. Additional ones are more work and we can add as needed.

Reply:
I was thinking about `XsrfOr` and `XsrfAnd` as well. At that point I wonder if it makes more sense to just use a function of type `Request -> Bool`. You get more power that way without needing to rely on a custom expression DSL.